samples/android_main.cc

```diff
@@ -23,23 +23,34 @@ extern int main(int argc, const char** argv);
 extern "C" JNIEXPORT JNICALL int Java_com_google_amber_Amber_androidMain(
     JNIEnv* env,
     jobject,
-    jstring args_jstring) {
+    jobjectArray args,
+    jstring stdoutFile,
+    jstring stderrFile) {
+  const char* stdout_file_cstr = env->GetStringUTFChars(stdoutFile, NULL);
+  const char* stderr_file_cstr = env->GetStringUTFChars(stderrFile, NULL);
+
   // Redirect std output to a file
-  freopen("/sdcard/amberlog.txt", "w", stdout);
-  freopen("/sdcard/amberlog.txt", "a", stderr);
+  freopen(stdout_file_cstr, "w", stdout);
+  freopen(stderr_file_cstr, "w", stderr);
+
+  env->ReleaseStringUTFChars(stdoutFile, stdout_file_cstr);
+  env->ReleaseStringUTFChars(stderrFile, stderr_file_cstr);
+
+  int arg_count = env->GetArrayLength(args);
```

Comment:
Nit but could you use `jsize` here and below.

Reply:
done